### ISSUE TITLE
Tool: measure throughput and set TRAIN_TOKEN_BUDGET before the base run — the LR anneal horizon is currently unverified

### ISSUE THREAD

**Issue by MatthewLacerda2** [MatthewLacerda2/TinyRefinementModel#156]
**Lane:** GPU (the measurement needs the card; the arithmetic and the PR do not).

## Why this blocks the base run

`TRAIN_TOKEN_BUDGET` is not a target for how good the model should be — it is the input that sets **how the learning rate anneals**. The chain, all in the repo today:

`TRAIN_TOKEN_BUDGET` (`trm/config.py:143`) → `resolve_decay_steps()` (`trm/train/schedules.py:17`) → `DECAY_STEPS` → the horizon of `optax.warmup_cosine_decay_schedule` (L35–39). This is #83's rule: *the LR anneal's horizon must match the run length.*

**Unset, it silently falls back to `_DEFAULT_DECAY_STEPS = 15000`** (`schedules.py:14`). With `TOKENS_PER_OPT_STEP = 64 × 2 × 2 × 512 = 131,072` (`config.py:124`), that default anneals the LR over **~1.97B tokens**, no matter how long the run actually goes. So:

- Run much *longer* than ~2B tokens → the LR is pinned at its floor for the remainder, and the tail of the run is wasted compute.
- Run much *shorter* → the run ends mid-cosine at a high LR, which is the classic never-finished-school symptom.

**And that second case is what has already been happening.** The runs that died at ~200M tokens were at ~1,526 optimizer steps — against `WARMUP_STEPS = 1000`. They had *just* cleared warmup and were roughly 3% into a 15,000-step anneal. "Undertrained" undersells it; those runs barely started their schedule.

Committing a week of GPU time without setting this is therefore not a small omission — it is starting the longest run in the project's history with an LR schedule aimed at a horizon nobody has checked.

## The gap

There is **no measured throughput anywhere in the repo** — no tokens/sec, no steps/sec, no wall-clock. Grep `docs/` and you get nothing. So the run length in tokens for any given wall-clock budget is currently unknown, which means `TRAIN_TOKEN_BUDGET` cannot be set from evidence.

## Do this

**1. Measure (on the GPU box).** The instrument already exists:

```bash
XLA_PYTHON_CLIENT_PREALLOCATE=false XLA_PYTHON_CLIENT_ALLOCATOR=platform \
  venv/bin/python -m instruments.bench_train_step --depth 8
```

It reports steps/sec and peak VRAM for both `loop` and `kernel` modes. **Use the `loop` number** — that is the mode which mimics the real training loop's per-step host sync. `kernel` is an upper bound the loop does not achieve.

**2. Convert — and mind the micro-step trap.** `bench_train_step --steps` counts **micro-steps**, not optimizer steps. `TOKENS_PER_OPT_STEP` is per *optimizer* step and already includes the ×64 accumulation. So:

```
tokens/sec = micro_steps_per_sec × BATCH_SIZE × 2 × MAX_SEQ_LEN     # = × 2048
```

Do **not** multiply micro-steps/sec by `TOKENS_PER_OPT_STEP` — that overstates throughput by 64×.

**3. Set the budget from wall-clock.** `tokens = tokens/sec × 3600 × (hours you intend to run)`. Round down; assume the card is not perfectly utilised for a week. Feed that to `TRAIN_TOKEN_BUDGET` so `DECAY_STEPS` matches the actual run, and sanity-check `resolve_decay_steps()` returns something comfortably above `WARMUP_STEPS = 1000` (it raises if not).

**4. Record it.** Non-novel measurement, so no `docs/findings/` entry — the PR is the record (rule 5). But the number must end up somewhere greppable, because every future planning decision wants it: add a throughput line to the hardware-reality paragraph in `CLAUDE.md`, stating the number, the depth and batch it was measured at, and the commit.

## Done when

- steps/sec (loop mode) and peak VRAM are measured at `--depth 8` and posted in a comment here
- tokens/sec and tokens-per-week are derived from it, with the ×2048 conversion shown
- `TRAIN_TOKEN_BUDGET` for the planned base run is chosen and stated
- the number is written into `CLAUDE.md`'s hardware paragraph in a small PR closing this issue

## Note on what this is *not*

This does not set a quality target. There is deliberately no "train until you hit X" bar here — the run trains as well as it can in the time given. This issue exists only so the LR schedule's horizon matches the run's actual length, which is a mechanical requirement, not a goal.

**Comment by MatthewLacerda2** [comment]
superseded-by #274, #269

What this issue asked for now exists mechanically:
- **Throughput is measured, per launch, on the real trainer:** the fit gate (#274) reports tok/s from the first logged row. On the RTX 2060 tonight, plain at 8 layers measured **4,407 tok/s** (plus arena peak 4113 MiB). The 4B refiner run's own log gave ~4.6k tok/s. Its premise, that no throughput is recorded anywhere, no longer holds, and `metrics.csv` now carries `wall_clock` (#258).
- **The budget can't be forgotten:** `make launch BUDGET=…` (#269) refuses without one, sets `TRAIN_TOKEN_BUDGET`, and derives the stop step on a checkpoint boundary. Relaunches adopt the run's recorded budget (`run_budget`).

What stays a human call and isn't closed by this: **how many tokens the next base run gets.** At 4,407 tok/s, 1B tokens ≈ 63h; 4B ≈ 10.5 days.
